### PR TITLE
SN 55 A basic budget page

### PR DESCRIPTION
**Pull request by theadzik**
* Copied `Budget`, `Category` (formerly `parent_category`), `Subcategory` (formerly `category`), and `Transaction` models from [this branch](https://github.com/infoshareacademy/jpydzr2-snaver-django/tree/SN-49-przeniesienie_podstawowych_modeli) and updated them slightly.
* Used `MockUser` for user Model: This needs to be replaced with a real User model when it's created.
* Created some basic tests to see if relations are working correctly.

### CODE REVIEW

File: snaver/admin.py [hunk 1]
```diff
@@ -1,3 +1,38 @@
...
 
-# Register your models here.
+from snaver.models import Budget
+from snaver.models import Category
+from snaver.models import MockUser
+from snaver.models import Subcategory
+from snaver.models import SubcategoryDetails
+from snaver.models import Transaction
+
+
```
Comment (KMdev94) [inline snaver/admin.py]: admin.site.register(Budget, Category ...)

File: snaver/models.py [hunk 2]
```diff
@@ -1,3 +1,106 @@
+from django.contrib.auth.models import AbstractUser
+from django.core.exceptions import ValidationError
+from django.core.validators import MinLengthValidator
 from django.db import models
+from django.utils import timezone
+from django.utils.translation import ugettext_lazy as _
 
-# Create your models here.
+
+class MockUser(AbstractUser):
```
Comment (KMdev94) [inline snaver/models.py]: Lepiej nazwac CustomUser

File: snaver/tests/tests_views.py [hunk 3]
```diff
@@ -0,0 +1,9 @@
+from django.test import Client
+from django.test import TestCase
+
+
+class CategoryView(TestCase):
+    def test_category_page_returns_200(self):
+        c = Client()
+        response = c.get('/budget/')
```
Comment (KMdev94) [inline snaver/tests/tests_views.py]: użyj reverse zamiast zahardkowanego adresu

File: snaver/models.py (hunk 2 above)
Comment (KMdev94) [reply snaver/models.py]: poprawić też w migracjach

**Review by joannaswiatek — APPROVED** [review]

**Pull request merged**